### ISSUE TITLE
Does Text classification utilize word embedding part?

### ISSUE THREAD

**Issue by ss87021456** [facebookresearch/fastText#469]
As the title, I wonder whether 
./fastText supervised
will utilize the fastText embedding such as skip-gram or CBOW when training the classifier or not.

**Comment by ss87021456** [comment]
Thanks! I've figured out this recently

**Comment by k4ni5h** [comment]
@ss87021456 , we tried `load_model `and `pretrained_vectors` flag but not able to train with custom emedding. Can you tell how did you trained the system with custom embedding ?

**Comment by ayushbits** [comment]
@ss87021456 Can you tell how did you pass pre-trained word embedding model as an input for classifcation?

**Comment by Bishnukuet** [comment]
> Thanks! I've figured out this recently

Hi, could you please tell me which model (Skip-gram or CBOW) is used when using supervised training of FastText?

**Comment by Celebio** [comment]
Hi @Bishnukuet,
When you use the supervised training, the word representation is neither skipgram nor cbow. The word vector you will obtain is tailored for the classification task you are training on.

Consider these tasks:
1) given a word, predict me which other words should go around
2) given a sentence with a missing word, find me the missing word
3) given a sentence, tell me which label corresponds to this sentence

For each of them, you have the following models : 1) skipgram, 2) cbow, 3) classification.
And for each of them, you will have word vectors in the latent space that fit best the tasks you are working on.

For 1) and 2), two words in the same context will have their vector representation close to each other. For 3), words that are most discriminative for a given label will be close to each other.

Best regards,
Onur

**Comment by junjun27** [comment]
> Hi @Bishnukuet,
> When you use the supervised training, the word representation is neither skipgram nor cbow. The word vector you will obtain is tailored for the classification task you are training on.
> 
> Consider these tasks:
> 
> 1. given a word, predict me which other words should go around
> 2. given a sentence with a missing word, find me the missing word
> 3. given a sentence, tell me which label corresponds to this sentence
> 
> For each of them, you have the following models : 1) skipgram, 2) cbow, 3) classification.
> And for each of them, you will have word vectors in the latent space that fit best the tasks you are working on.
> 
> For 1) and 2), two words in the same context will have their vector representation close to each other. For 3), words that are most discriminative for a given label will be close to each other.
> 
> Best regards,
> Onur

Hi,
Thanks for your information, it's really nice to know that. 
Could you explain how to get word embedding for each word after the supervised training? I trained a supervised model with my own dataset, and finally got a model.bin.  I want to know how to get word embedding from this model.
Thanks a lot in advance.

Best regards,
Jun

**Comment by RicardoSouzaMoura** [comment]
The way I understood the supervised training generate two files: .bin and .vec
.bin is the model and .vec is the word embedding

**Comment by ziorufus** [comment]
Thank you @Celebio for your great answer. I have another question about this topic: how does the training/classification actually work for the supervised task? Does the training actually generate also a vector for each label?
For what I understood, when a sentence is given to the system, FastText get the vectors for each word (or n-gram) and calculate the average. Then how does the system choose the correct label? With what the sentence vector calculated from the input text is compared?
